src/protocol/bigreq.rs

```diff
@@ -32,26 +33,41 @@ pub const X11_EXTENSION_NAME: &str = "BIG-REQUESTS";
 /// send the maximum version of the extension that you need.
 pub const X11_XML_VERSION: (u32, u32) = (0, 0);
 
-/// Opcode for the Enable request
-pub const ENABLE_REQUEST: u8 = 0;
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct EnableRequest;
+impl EnableRequest {
+    /// Opcode for the Enable request
+    pub const fn opcode() -> u8 { 0 }
+    /// Serialize this request into bytes for the provided connection
+    #[allow(unused)]
```

Comment:
What's unused here?

From a quick check: This only happens with requests from `xproto` that do not use the `conn` argument at all.
Would it make sense to just remove the `conn` argument from the requests in xproto? That sounds like a relatively simple check in the code generator.

Another random idea that I am not to sure about: Replace the `conn` argument in extensions with a `major_opcode` argument and move the extension check to the caller? Is there any use case where this might be helpful?

Reply:
The reason I didn't do this in the first place is that having differing signatures for differing messages makes it harder to split this off into another trait to enable something like #212.

Reply:
How about adding `let _ = conn;` as an alternative to the `extension_information(X11_EXTENSION_NAME)?` dance? That should also silence the warning

Reply:
Done.